### PR TITLE
refactor(experimental): added unsafe f64 type

### PR DESCRIPTION
**Pull request by buffalojoec**
refactor(experimental): added unsafe f64 type

### CODE REVIEW

File: packages/rpc-core/src/rpc-methods/common.ts [hunk 1]
```diff
@@ -29,6 +29,15 @@ export type Slot = U64UnsafeBeyond2Pow53Minus1;
...
 export type U64UnsafeBeyond2Pow53Minus1 = bigint;
 
+// FIXME(solana-labs/solana/issues/30341)
+// <https://stackoverflow.com/questions/45929493/node-js-maximum-safe-floating-point-number/57225494#57225494>
+// Beware that floating-point value precision can vary widely:
+// - For precision of 1 decimal place, anything above 562949953421311
+// - For precision of 2 decimal places, anything above 70368744177663
+// can be truncated or rounded because of a downcast to JavaScript `number` between your calling
+// code and the JSON-RPC transport.
+export type F64UnsafeSingleDecimalPointBeyond2Pow53Over16Minus1 = number;
```
Comment (steveluscher) [inline packages/rpc-core/src/rpc-methods/common.ts:39]: I mean, I love it, but probably better to just be like `F64UnsafeSeeCodeCommentForWhy` or something :)

**Review by steveluscher — APPROVED** [review]
giphy-1

File: packages/rpc-core/src/rpc-methods/common.ts (hunk 1 above)
Comment (buffalojoec) [reply packages/rpc-core/src/rpc-methods/common.ts:39]: I figured the longer the name the more obvious of a warning it is. Your suggestion is also quite long

Comment (mcintyre94) [reply packages/rpc-core/src/rpc-methods/common.ts:39]: `__F64_UNSAFE___DO_NOT_USE_OR_YOU_WILL_BE_FIRED`

**Review by mcintyre94 — APPROVED** [review]

File: packages/rpc-core/src/rpc-methods/common.ts (hunk 1 above)
Comment (buffalojoec) [reply packages/rpc-core/src/rpc-methods/common.ts]: I went with `F64UnsafeSeeDocumentation`

**Pull request merged**